### PR TITLE
Add blockchain compatibility

### PR DESCRIPTION
**Pull request by AndrewBoyarsky**

### CODE REVIEW

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/BlockchainImpl.java [hunk 1]
```diff
@@ -476,8 +484,37 @@ public int getTransactionCount(long accountId, byte type, byte subtype) {
 
 
     @Override
-    public Set<Long> getBlockGenerators(int startHeight) {
-        return lookupBlockDao().getBlockGenerators(startHeight);
+    public Set<Long> getBlockGenerators(int limit) {
+        int startHeight = getHeight() - 10_000;
```
Comment (yuriy-larin-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/BlockchainImpl.java]: A little bit strange constant = 10_000

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/BlockDaoImpl.java [hunk 2]
```diff
@@ -350,14 +315,16 @@ public int getBlockCount(long accountId) {
...
+        catch (SQLException e) {
             throw new RuntimeException(e.toString(), e);
         }
     }
 
     @Override
-    public List<Long> getBlockIdsAfter(long blockId, int limit, List<Long> result) {
+    public List<Long> getBlockIdsAfter(long blockId, int limit) {
         // Search the database
+        List<Long> result = new ArrayList<>(limit);
```
Comment (yuriy-larin-firstbridge) [inline apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/BlockDaoImpl.java]: Default DataSource for main db is fetched here in all cases. Is it correct for now?

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/ActiveGenerators.java [hunk 3]
```diff
@@ -53,7 +53,7 @@ public ActiveGenerators(Blockchain blockchain) {
     @PostConstruct
     public void init() {
         if (!generatorsInitialized) {
-            activeGeneratorIds.addAll(blockchain.getBlockGenerators(Math.max(1, blockchain.getHeight() - 10000)));
+            activeGeneratorIds.addAll(blockchain.getBlockGenerators(250));
```
[the remark this answers is not in the document]
Comment (AndrewBoyarsky) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/ActiveGenerators.java]: Moved to constants

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/BlockchainImpl.java (hunk 1 above)
Comment (AndrewBoyarsky) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/BlockchainImpl.java]: Moved to constants

File: apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/BlockDaoImpl.java (hunk 2 above)
Comment (AndrewBoyarsky) [reply apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/db/BlockDaoImpl.java]: Yes, its correct. See getBlockIdsAfter in BlockchainImpl. We extract block ids from index and then forward further search to current db, if required.

**Pull request merged**